src/quo2/components/selectors/selectors.cljs

```diff
@@ -0,0 +1,129 @@
+                                          (get-color value disabled?))})
+
+        :accessibility-label (str "checkbox-" (if value "on" "off"))
+        :accessibility-role  :checkbox}
+       (when value
+         [rn/view {:style {:height 20
+                           :width 20}}
+          [icons/icon :main-icons2/check
+           {:size 20
+            :color (colors/alpha colors/white (if disabled? 0.3 1))}]])]])))
+
+(defn radio-view [props]
+  (let [{:keys [value onChange disabled? containerStyle]} (bean/bean props)]
+    (reagent/as-element
+     [rn/touchable-without-feedback
+      {:on-press (when (and onChange (not disabled?)) onChange)}
+      [rn/view
+       {:style        (merge
+                       (js->clj containerStyle)
+                       {:height 20
+                        :width 20
+                        :border-radius 40
+                        :border-width 1
+                        :border-color (get-color value disabled?)
+                        :background-color (colors/theme-colors colors/white
+                                                               (colors/alpha colors/neutral-80 0.4))})
+        :accessibility-label (str "radio-" (if value "on" "off"))
+        :accessibility-role  :checkbox}
+
+       [rn/view {:style {:margin-left :auto
+                         :height 14
+                         :width 14
+                         :background-color (when value (get-color value disabled?))
+                         :border-radius 40
+                         :margin-right :auto
+                         :margin-top :auto
+                         :margin-bottom :auto}}]]])))
+
+(defn toggle-view [props]
+  (let [{:keys [value onChange disabled? containerStyle]} (bean/bean props)]
+    (reagent/as-element
+     [rn/touchable-without-feedback
+      {:on-press (when (and onChange (not disabled?)) onChange)}
+      [rn/view
+       {:style        (merge
+                       (js->clj containerStyle)
+                       {:height 20
+                        :width 30
+                        :border-radius 40
+                        :background-color (get-color value disabled?)})
+        :accessibility-label (str "toggle-" (if value "on" "off"))
+        :accessibility-role  :checkbox}
+       [rn/view {:style {:margin-left (if value 12 2)
+                         :height 16
+                         :width 16
+                         :background-color (colors/alpha colors/white
+                                                         (if (theme/dark?)
+                                                           (if disabled? 0.3 1)
+                                                           1))
+                         :border-radius 40
+                         :margin-right :auto
+                         :margin-top :auto
+                         :margin-bottom :auto}}]]])))
+
+(def radio (reagent/adapt-react-class (react/memo radio-view)))
```

Comment:
Please, correct me if I'm wrong @J-Son89, but I think `React.memo` only works reliably if we're 100% sure React can skip rendering based on shallow comparisons of the props.

Let's take the `toggle-view` component as an example, it depends on `theme/dark?`, which depends on the ratom `quo.theme/theme`.

I honestly don't understand how the component renders correctly if the outside state changes (switch dark/light modes).

Reply:
yep, you are right. I tested this out just to be 100%. Should we remove the react/memo so? or look for ways to optimise this? WDYT ? 🤔

Reply:
I'm just generally wary of things I don't understand. In this case, I really can't understand why `React.memo` works reliably with external state affecting how colors are calculated. Maybe if it works is because the entire tree is being evicted from the cache, so in reality the memo does nothing and React is always re-rendering. If you tested and it works it looks fine by me, but I'd avoid the memo unless we profiled these components and found performance issues. It's up to you I guess `¯\_(ツ)_/¯`, unless other people have something to comment on.

Reply:
So I guess the best we could do now is remove the cache wrapper. If we want to optimize the app, I think the best we can do is profile and identify the bottlenecks. If we find that such components are problematic we can get always get back to the drawing table. WDYT?

Reply:
Yeah I think for now there's no need to memoize unless it becomes an issue. 
Realistically this potential ui bug from memoizing would only happen if there is other selectors on the page where we toggle the theme. So it's pretty much safe to have memoization as long as the screen is designed accordingly.

So I will remove for now and if needs be we can improve this later if necessary.